### PR TITLE
CBO: Travel automation full 

### PR DESCRIPTION
**Pull request by cleargif**
#### What
This branch contains all the travel automation work.
- `Helpers.Block.ExpenseBlock`
-- A constructor class to act as a controller for `ExpenseBlocks` 
-- The controllers handles the state + event actions on each `ExpenseBlock`
- `Helpers.API.Core`
-- Helper to handle all ajax calls from the FE.
- `Helpers.API.Distance`
-- Helper to handle distance lookups
- `Helper.API.Establishments`
-- Helper to handle establishment data loading
-- Helper methods to interpolate the data into HTML elements
- `Modules.DuplicateExpenseCtrl`
-- A controller to manage expense duplication
-- handles the differences between AGFS and LGFS expenses
- `Helpers.FormControls`
-- Methods to interpolate the data into HTML elements
-- An API to create `input`, `select` and `option` elements
- The View amends on the expense page
- The View amends on the BE
- The BE code to support the API, new data, specs and features
- Features
-- `advocate_expense_page.feature` and `litigator_expense_page.feature` expense page state and JS integration
- Jasmine tests
-- The new modules have a decent unit test coverage 
-- `helpers-sidebar_spec.js`, `modules-DuplicateExpenseCtrl_spec.js`, `modules-sidebar_spec.js`, `Helpers.API.Core_spec.js`, `Helpers.API.Distance_spec.js`, `Helpers.API.Establishment_spec.js`, `Helpers.FormControls_spec.js`
- `Module.ExpensesDataTable.js`
-- `columnDef` updated to force the space and better display the travel distance info on the case summaty

#### Tickets
CBO-151, CBO-202, CBO-153 + sub tasks 

#### Why
To automatically calculate the distance for `Bike travel` or `Car travel`
To give more contextual information about the expense reason and location
This feature is enabled on LGFS only.

### CODE REVIEW

File: app/assets/javascripts/application.js [hunk 1]
```diff
@@ -87,7 +87,7 @@ if (!String.prototype.supplant) {
    * Cocoon call back to init features once they have been
    * interted into the DOM
    */
-  $('#fixed-fees, #misc-fees, #disbursements, #expenses, #documents').on('cocoon:after-insert', function(e, insertedItem) {
+  $('#fixed-fees, #misc-fees, #disbursements, #documents').on('cocoon:after-insert', function(e, insertedItem) {
```
Comment (colinbruce) [inline app/assets/javascripts/application.js:90]: Are you not using cocoon for expenses anymore?  Still reading... ignore if addressed later!

File: spec/vcr_helper.rb [hunk 2]
```diff
@@ -16,8 +17,12 @@
 
   # Ignore requests other than to the API endpoints and LAA fee calculator
```
Comment (colinbruce) [inline spec/vcr_helper.rb]: Should probably update to
```ruby
#Ignore requests other than to the API endpoints, LAA fee calculator and google maps
```
for clarity

**Review by colinbruce — APPROVED** [review]
Minor observations only so far

File: app/assets/javascripts/application.js (hunk 1 above)
Comment (cleargif) [reply app/assets/javascripts/application.js:90]: This is a specific method to init any type ahead components after a cocoon insert. 

Expenses does not have `typeahead` selects anymore

File: spec/vcr_helper.rb (hunk 2 above)
Comment (cleargif) [reply spec/vcr_helper.rb]: 👍  will do..

**Comment by cleargif** [comment]
Moved to PR:  #2500

**Pull request closed without merging**